websocket/platform/sim.w

```diff
@@ -0,0 +1,62 @@
+bring cloud;
+bring util;
+bring "../commons/api.w" as api;
+
+interface StartWebSocketApiResult {
+  inflight close(): inflight(): void;
+  inflight url(): str;
+}
+
+pub class WebSocket_sim impl api.IWebSocket {
+  var connectFn: inflight(str): void;
+  var disconnectFn: inflight(str): void;
+  var messageFn: inflight(str, str): void;
+  bucket: cloud.Bucket;
+
+  new(props: api.WebSocketProps) {
+    this.connectFn = inflight () => {};
+    this.disconnectFn = inflight () => {};
+    this.messageFn = inflight () => {};
+    this.bucket = new cloud.Bucket();
+  }
+
+  pub onConnect(handler: inflight(str): void): void {
+    this.connectFn = handler;
+  }
+  pub onDisconnect(handler: inflight(str): void): void {
+    this.disconnectFn = handler;
+  }
+  pub onMessage(handler: inflight(str, str): void): void {
+    this.messageFn = handler;
+  }
+
+  pub initialize() {
+    new cloud.Service(inflight () => {
+      let res = WebSocket_sim._startWebSocketApi(this.connectFn, this.disconnectFn, this.messageFn);
+      this.bucket.put("url.txt", res.url());
```

Comment:
I think we could use a `sim.State` here.

Reply:
I agree, here is a reference how it is being used 

https://github.com/winglang/winglibs/blob/main/containers/workload.sim.w

Reply:
see above

Reply:
I think @skyrpex means to use `sim.State` for the url (instead of the bucket)